src/cvc.mli

```diff
@@ -0,0 +1,33 @@
+module RSA :
+sig
+  module Public :
+  sig
+    type t =
+      { n: Z.t
+      ; e: Z.t
+      }
+    [@@deriving ord,yojson,eq,show]
+  end
+end
+
+module ECDSA :
+sig
+  module Public :
+  sig
+    type t =
+      { modulus : Z.t
+      ; coefficient_a : Z.t
+      ; coefficient_b : Z.t
+      ; base_point_g : Z.t
+      ; base_point_r_order : Z.t
+      ; public_point_y : Z.t
+      ; cofactor_f : Z.t
+      }
+    [@@deriving ord,yojson,eq,show]
+  end
+end
+
+type t =
```

Comment:
Also, the `decode` functions are pretty much overlays of the master `decode` function, checking to see that the return value is correctly wrapped. It would be better to modify the master `decode` function to accept an OID as a parameter and terminate parsing if the OID it finds doesn't match the one given, but that sort of modification would have taken far longer and from what I understand speed isn't the highest priority for this feature at this stage.